packages/dapp-connector/src/WalletApi/Cip30Wallet.ts

```diff
@@ -1,7 +1,71 @@
 import { APIErrorCode, ApiError } from '../errors';
+import {
+  Bytes,
+  Cbor,
+  Cip30WalletApi,
+  Cip95WalletApi,
+  Paginate,
+  WalletApi,
+  WalletApiExtension,
+  WalletMethod
+} from './types';
+import { Cardano } from '@cardano-sdk/core';
 import { Logger } from 'ts-log';
 import { RemoteAuthenticator } from '../AuthenticatorApi';
-import { WalletApi } from './types';
+
+export const Cip30WalletApiMethodNames: (keyof Cip30WalletApi)[] = [
+  'getNetworkId',
+  'getUtxos',
+  'getCollateral',
+  'getBalance',
+  'getUsedAddresses',
+  'getUnusedAddresses',
+  'getChangeAddress',
+  'getRewardAddresses',
+  'signTx',
+  'signData',
+  'submitTx'
+];
+export const Cip95WalletApiMethodNames: (keyof Cip95WalletApi)[] = ['getActivePubStakeKeys', 'getPubDRepKey'];
+export const WalletApiMethodNames: WalletMethod[] = [...Cip30WalletApiMethodNames, ...Cip95WalletApiMethodNames];
+
+/**
+ * Wrap the proxy API object with a regular javascript object to avoid interop issues with some dApps.
+ *
+ * Only return the allowed API methods.
+ */
+const wrapAndEnableApi = (walletApi: WalletApi, allowedApiMethods: WalletMethod[]): WalletApi => {
+  const objectApi: WalletApi = {
+    getActivePubStakeKeys: () => walletApi.getActivePubStakeKeys(),
+    getBalance: () => walletApi.getBalance(),
+    getChangeAddress: () => walletApi.getChangeAddress(),
+    getCollateral: (params?: { amount?: Cbor }) => walletApi.getCollateral(params),
+    getNetworkId: () => walletApi.getNetworkId(),
+    getPubDRepKey: () => walletApi.getPubDRepKey(),
+    getRewardAddresses: () => walletApi.getRewardAddresses(),
+    getUnusedAddresses: () => walletApi.getUnusedAddresses(),
+    getUsedAddresses: (paginate?: Paginate) => walletApi.getUsedAddresses(paginate),
+    getUtxos: (amount?: Cbor, paginate?: Paginate) => walletApi.getUtxos(amount, paginate),
+    signData: (addr: Cardano.PaymentAddress | Bytes, payload: Bytes) => walletApi.signData(addr, payload),
+    signTx: (tx: Cbor, partialSign?: Boolean) => walletApi.signTx(tx, partialSign),
+    submitTx: (tx: Cbor) => walletApi.submitTx(tx)
+  };
+
+  const enabledApi = allowedApiMethods.reduce(
+    (api, methodName) => ({ ...api, [methodName]: objectApi[methodName] }),
```

Comment:
Please convert this to something more like
```
const enableApi = Object.fromEntries(
  Object.entries(objectApi)
  .filter((methodName, _) => allowedMethods.includes(methodName));
```
to remove the reduce-spread pattern